src/cmds/categories/create.js

```diff
@@ -0,0 +1,127 @@
+const fetch = require('../../lib/fetch');
+const { cleanHeaders, handleRes } = require('../../lib/fetch');
+const { debug } = require('../../lib/logger');
+
+module.exports = class CategoriesCreateCommand {
+  constructor() {
+    this.command = 'categories:create';
+    this.usage = 'categories:create <title> [options]';
+    this.description = 'Create a category with the specified title and guide in your ReadMe project';
+    this.category = 'categories';
+    this.position = 2;
+
+    this.hiddenargs = ['title'];
+    this.args = [
+      {
+        name: 'key',
+        type: String,
+        description: 'Project API key',
+      },
+      {
+        name: 'version',
+        type: String,
+        description: 'Project version',
+      },
+      {
+        name: 'title',
+        type: String,
+        defaultOption: true,
+      },
+      {
+        name: 'categoryType',
+        type: String,
+        description: 'Category type, must be guide or reference',
+      },
+      {
+        name: 'preventDuplicates',
+        type: Boolean,
+        description:
+          'Prevents the creation of a new category if their is an existing category with a mathcing `categoryType` and exact matching `title`',
+      },
+    ];
+  }
+
+  async run(opts) {
+    const { categoryType, title, key, version, preventDuplicates } = opts;
+    debug(`command: ${this.command}`);
+    debug(`opts: ${JSON.stringify(opts)}`);
+
+    if (!key) {
+      return Promise.reject(new Error('No project API key provided. Please use `--key`.'));
+    }
+
+    if (!title) {
+      return Promise.reject(new Error(`No title provided. Usage \`${config.get('cli')} ${this.usage}\`.`));
+    }
+
+    if (categoryType !== 'guide' && categoryType !== 'reference') {
+      return Promise.reject(new Error('`categoryType` must be guide or reference.'));
+    }
+
+    const selectedVersion = await getProjectVersion(version, key, false);
+
+    debug(`selectedVersion: ${selectedVersion}`);
+
+    function getNumberOfPages() {
+      return fetch(`${config.get('host')}/api/v1/categories?perPage=20&page=1`, {
+        method: 'get',
+        headers: cleanHeaders(key, {
+          'x-readme-version': selectedVersion,
+          Accept: 'application/json',
+        }),
+      }).then(res => {
+        return Math.ceil(res.headers.get('x-total-count') / 20);
+      });
+    }
+
+    const allCategories = [].concat(
+      ...(await Promise.all(
+        Array.from({ length: await getNumberOfPages() }, (_, i) => i + 1).map(async page => {
+          return fetch(`${config.get('host')}/api/v1/categories?perPage=20&page=${page}`, {
+            method: 'get',
+            headers: cleanHeaders(key, {
+              'x-readme-version': selectedVersion,
+              Accept: 'application/json',
+            }),
+          }).then(res => handleRes(res));
+        })
+      ))
+    );
+
+    async function matchCategory() {
+      return allCategories.filter(category => {
```

Comment:
Small performance suggestion: switching to [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) here will have it stop iterating once it's found what it's looking for (plus, it returns a boolean!):

```suggestion
      return allCategories.some(category => {
```

Reply:
@kanadgupta Agree from a performance standpoint, but we actually want to return the category object as some of it's attributes such as the `id` are used in the response when a duplicate is found.

Reply:
Ah my apologies, I missed the part where we use the found object's attributes. Could you switch to [`Array.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead? That way we get the aforementioned performance benefits and can use the object properly.

```suggestion
      return allCategories.find(category => {
```